plugins/catalog-import/src/components/StepPrepareCreatePullRequest/CheckboxField.tsx

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Checkbox, FormControlLabel, FormHelperText } from '@material-ui/core';
+import React from 'react';
+
+type Props<TFieldValue extends string> = {
+  name: TFieldValue;
+  label: React.ReactNode;
+  inputRef:
+    | ((instance: HTMLInputElement | null) => void)
+    | React.RefObject<HTMLInputElement>
+    | null
+    | undefined;
+  onChange?: (
+    event: React.ChangeEvent<HTMLInputElement>,
+    checked: boolean,
+  ) => void;
+  helperText?: React.ReactNode | string;
+};
+
+export const CheckboxField = <TFieldValue extends string>({
```

Comment:
Not sure if we really need this extra component. There are some quite complex type definitions and I'm not sure if it's worth doing that. Why not add the structure to the defaults.tsx? The Autocomplete component is special because it is not compatible to the form library by default which I would expect the checkbox to be.

Reply:
The main reason I did this was to to ecapsulate additional disclaimer text. I originally tried putting in an warning icon and also wrapping the text under the label, but the alignment starts looking funny. So I ended up falling back to just planning it under the checkbox and label altogether to avoid spending too much time styling. But yeah, now it seems a bit unnecessary for a different component.